theseus/core/robust_loss.py

```diff
@@ -50,3 +50,64 @@ def _evaluate_impl(x: torch.Tensor, radius: torch.Tensor) -> torch.Tensor:
     @staticmethod
     def _linearize_impl(x: torch.Tensor, radius: torch.Tensor) -> torch.Tensor:
         return torch.sqrt(radius / torch.max(x, radius) + _LOSS_EPS)
+
+
+class HingeLoss(RobustLoss):
+    @staticmethod
+    def _evaluate_impl(x: torch.Tensor, radius: torch.Tensor) -> torch.Tensor:
+        return torch.where(x > radius, torch.sqrt(x) - torch.sqrt(radius), _LOSS_EPS)
+
+    @staticmethod
+    def _linearize_impl(x: torch.Tensor, radius: torch.Tensor) -> torch.Tensor:
+        return torch.where(x > radius, 1.0 / (2 * torch.sqrt(x) + _LOSS_EPS), 0.0)
+
+
+class GNCRobustLoss(RobustLoss):
```

Comment:
`class GNCRobustLoss(RobustLoss, abc.ABC)` since this also has abstract methods and the class cannot be instantiated.